### PR TITLE
LPD-22281 Create Adyen Payment Integration Client Extension

### PR DESCRIPTION
**Pull request by ghost**
https://liferay.atlassian.net/browse/LPD-22281

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:stop

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

File: workspaces/liferay-adyen-workspace/client-extensions/liferay-adyen-payment-integration/client-extension.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    oAuth2ApplicationExternalReferenceCode: liferay-adyen-payment-integration-oauth-application-user-agent
+    paymentIntegrationType: 4
+    paymentIntegrationTypeSettings:
+        apiKey: AQEshmfxL4jOYxVCw0m/n3Q5qf3VZIJLCIFLQTEPl6IyMyeFdrvMGYnwdW52sTQQwV1bDb7kfNy1WIxIIkxgBw==-zPLFeWNsZcjcpVBAa4OPgzmBMPODfQ0Gzq2KIuon658=-CKq,[,>4&W79Fwun
+        environment: TEST
+        hmacSignature: 94EF50ED792BCA2417617298E91F23B0086DF763972734EEE01F66FCB8B900D8
+        merchantAccount: Liferay491ECOM
+        themeId: 6603dffc-8a21-4ecd-adc4-d0b6981ba967
+        webhookPassword: pippo
+        webhookUsername: test
```
Comment (luca-pellizzon) [inline workspaces/liferay-adyen-workspace/client-extensions/liferay-adyen-payment-integration/client-extension.yaml:15]: need to remove these values

File: workspaces/liferay-adyen-workspace/client-extensions/liferay-adyen-payment-integration/src/main/java/com/liferay/adyen/CaptureRestController.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+					).getString(
+						"id"
+					),
+					_log);
+
+				paymentStatus = "8";
+			}
+		}
+		catch (Exception exception) {
+			errorMessages = ExceptionUtils.getStackTrace(exception);
```
Comment (luca-pellizzon) [inline workspaces/liferay-adyen-workspace/client-extensions/liferay-adyen-payment-integration/src/main/java/com/liferay/adyen/CaptureRestController.java:84]: don't like this error handling, what if the integration itself throws an error like the sessionResultResponse contains errors, we are not  using/sending them back

Comment (luca-pellizzon) [reply workspaces/liferay-adyen-workspace/client-extensions/liferay-adyen-payment-integration/src/main/java/com/liferay/adyen/CaptureRestController.java:84]: this is for all methods not only the capture

File: workspaces/liferay-adyen-workspace/client-extensions/liferay-adyen-payment-integration/src/main/java/com/liferay/adyen/WebhookRestController.java [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+import reactor.core.publisher.Mono;
+
+/**
+ * @author Crescenzo Rega
+ */
+@RequestMapping("/api")
+@RestController
+public class WebhookRestController extends BaseRestController {
+
+	@PostMapping("/webhooks/notifications")
```
Comment (luca-pellizzon) [inline workspaces/liferay-adyen-workspace/client-extensions/liferay-adyen-payment-integration/src/main/java/com/liferay/adyen/WebhookRestController.java:57]: why `/webhooks/notifications`?

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/integration/FunctionCommercePaymentIntegration.java [hunk 4]
```diff
@@ -229,6 +236,13 @@ private JSONObject _getPayloadJSONObject(
 		String className = _portal.getClassName(
 			commercePaymentEntry.getClassNameId());
 
+		String sessionResult = ParamUtil.getString(
+			httpServletRequest, "sessionResult");
+
+		if (!Validator.isBlank(sessionResult)) {
+			commercePaymentEntryJSONObject.put("sessionResult", sessionResult);
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/integration/FunctionCommercePaymentIntegration.java:243]: make this as much generic as possible, by sending all httpServletRequest params and not a specific one

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:report:2571720022

**Comment by CrescenzoRegaLF** [comment]
ci:report:2571720022

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:report:2572744180

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
@luca-pellizzon 
https://github.com/brianchandotcom/liferay-portal/pull/150839#issuecomment-2145632548
https://github.com/brianchandotcom/liferay-portal/pull/150982

**Pull request closed without merging**